calico/felix/fetcd.py

```diff
@@ -334,6 +335,54 @@ def __init__(self, config, etcd_api, status_reporter, hosts_ipset):
         self.last_rate_log_time = monotonic_time()
         # Register for events when values change.
         self._register_paths()
+        self.estimated_host_count()
+        self._usage_count_greenlet = gevent.Greenlet(
+            self._periodically_usage_count
+        )
+        # self._usage_count_greenlet.link_exception(self._on_worker_died)
+
+    # def _on_worker_died(self, watch_greenlet):
+    #     """
+    #     Greenlet: spawned by the gevent Hub if the etcd watch loop ever
+    #     stops, kills the process.
+    #     """
+    #     _log.critical("Worker greenlet died: %s; exiting.", watch_greenlet)
+    #     sys.exit(1)
+
+    def estimated_host_count(self):
+        return len(self.ipv4_by_hostname)
+
+    @logging_exceptions
+    def _periodically_usage_count(self):
+        """
+        Greenlet: periodically report the cluster existence to
+        projectcalico.org.
+
+        :return: Does not return, unless reporting disabled.
+        """
+        _log.info("Started usage count thread. Waiting for config.")
+        self.configured.wait()
+        interval = 86400 # Once every 24 hours
+        # gevent.sleep(15) # Sleep for 15 seconds to give estimated_host_count time
+        _log.debug("Usage count interval: %s", interval)
+
+        while True:
+            try:
+                config = self._config
+                calico_version = str(pkg_resources.require("calico")[0].version)
+                _log.info("calico_version = %s", calico_version)
+                felix_count = self.estimated_host_count()
+                _log.info("felix_count = %s", felix_count)
+                usage_count(calico_version, config.HOSTNAME, config.CLUSTER_GUID, felix_count, config.CLUSTER_TYPE)
+            except:
```

Comment:
This should be `except Exception:` so that we don't catch `SystemExit` an `KeyboardInterrupt`.  Those _should_ kill the app.